pkg/applier/applier.go

```diff
@@ -246,8 +246,15 @@ func (h *eventHandler) processApplyEvent(ctx context.Context, e event.ApplyEvent
 		switch e.Error.(type) {
 		case *applyerror.UnknownTypeError:
 			unknownTypeResources[id] = struct{}{}
+
+			if e.Resource != nil {
```

Comment:
You're right, I'll try a different approach then and add an e2e test to verify it

Reply:
You can make changes to cli-utils, if necessary. 

That said, the last time I looked, the applier strips the source path annotation from the object before applying. So you might have to jump through some hoops if that's what you wanted. 

Also, it's a little unclear what object should be returned. Do you want the object from source, the object applied, or the final object after apply?

When the apply fails, you only have the first two, not the current state on the cluster.

Reply:
But if you only want the object from source, that's available to the supervisor already, because it passes it into the apply. You could pull it from there if you need.

Reply:
@sdowell what do you think?

Reply:
I think there are several potential solutions, so I don't want to be too prescriptive of the solution without doing some prototyping myself to compare.

I think Karl's suggestion is probably top of list for what I'd try out, as I expect it will involve the most minimal amount of change.

If we're using the source path annotation on the object, then I do think we want to be using the object from source. My initial thinking however was that the file path would be passed along as a variable into the applier (e.g. refactoring the translation that happens [here](https://github.com/GoogleContainerTools/kpt-config-sync/blob/v1.18.0-rc.3/pkg/parse/updater.go#L177)). I think using the hydrated annotation makes it harder to understand how the information is being passed around the code, although it probably requires less refactoring to just use the annotation.